biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -90,8 +89,7 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 				Discover discover = findDiscoveryMode(beansResource);
 
 				jar.getResources()
-					.values()
-					.forEach(r -> discoverPerResource.put(r, discover));
+					.forEach((k, r) -> resourceDiscovers.add(new Tuple(k, r, discover)));
```

Comment:
Since you only care about class resources (endsWith(".class")), perhaps you can filter non-class resources out here? To make the list shorter when processing below?